inductiva/fluids/scenarios/coastal_area/coastal_area.py

```diff
@@ -75,6 +79,96 @@ def from_text_file(
+        by the `initial_roughness` and `roughness_factor` parameters, that set
+        the initial range of randomness and the rate at which it decreases over
+        iterations of the algorithm, respectively. The size of the resultnig
+        grid is `n`, where `n` is the smallest integer that satisfies
+        `2^n + 1 >= max(x_num, y_num)`.
+        2. The depths of the square grid are interpolated to a grid of shape
+        `(x_num, y_num)`.
+
+        The depths of the corners of the grid are chosen according to a maximum
+        depth value `max_depth` and a percentile of the domain that must be
+        above water `percentile_above_water`. The corners on the lower x
+        boundary (East) are assumed to be below water (i.e. have 0 < depths <
+        `max_depth`). The corners on the upper x boundary (West) are assumed to
+        be above water (i.e. have - `max_depth` < depths < 0).
+        
+        Args:
+            x_range: The range of x values, in meters.
+            y_range: The range of y values, in meters.
+            x_num: Number of grid points in the x direction.
+            y_num: Number of grid points in the y direction.
+            max_depth: Maximum depth value, in meters.
+            initial_roughness: Initial roughness value, in meters. Controls the
+              initial range of randomness of the Diamond-Square algorithm.
+            roughness_factor: Roughness factor. Must be between 0 and 1.
+              Controls the rate at which the range of randomness of the
+              Diamond-Square algorithm decreases over iterations.
+            percentile_above_water: Percentile of the depths that must be above
+              water. Must be between 0 and 100.
+        """
+
+        corner_values = [
+            random.uniform(0, max_depth),
+            random.uniform(0, max_depth),
+            random.uniform(-max_depth, 0),
+            random.uniform(-max_depth, 0),
+        ]
+
+        # Determine the minimum n such that 2^n + 1 >= max(x_num, y_num).
+        n = int(math.log2(max(x_num, y_num) - 1)) + 1
+
+        size_square = 2**n + 1
+
+        depths = diamond_square.create_random_array(
+            size=size_square,
+            corner_values=corner_values,
+            initial_roughness=initial_roughness,
+            roughness_factor=roughness_factor)
+
+        # Adjust depths to ensure that a given percentage of the domain is above
+        # sea level (depth < 0).
+        percentile_under_water = np.percentile(depths, percentile_above_water)
+        depths -= percentile_under_water
+
+        x_square = np.linspace(*x_range, size_square)
+        y_square = np.linspace(*y_range, size_square)
+
+        x_square, y_square = np.meshgrid(x_square, y_square, indexing="ij")
+
+        x = np.linspace(*x_range, x_num)
+        y = np.linspace(*y_range, y_num)
+
+        x, y = np.meshgrid(x, y, indexing="ij")
+
+        depths = scipy.interpolate.griddata(
```

Comment:
A side effect of this is that it smooths out the bathymetry. It's not a problem, but to get more interesting simulation cases, we probably need to set higher roughness factor (see my comment on the video).